File: 8kyu/(8 kyu) Powers of 2/(8 kyu) Powers of 2.py
# 1 Simple solution
def powers_of_two(n):
    res = []
    for i in range(n + 1):
        res.append(2 ** i)
    return res

# 2 Optimized solution
def powers_of_two(n):
    return [2 ** i for i in range(n + 1)]

# 3 Clever solution
def powers_of_two(n):
    return [1 << i for i in range(n + 1)]

# 4 Coding golf
powers_of_two=lambda n:[1<<i for i in range(n+1)]

File: 8kyu/(8 kyu) Powers of 2/(8 kyu) Powers of 2.cs
// 1 Simple solution
using System.Numerics;

public class Kata
{
  public static BigInteger[] PowersOfTwo(int n)
  {
    BigInteger[] result = new BigInteger[n + 1];
    for (int i = 0; i <= n; i++)
    {
      result[i] = BigInteger.Pow(2, i);
    }
    return result;
  }
}

// 2 Optimized solution
using System.Linq;
using System.Numerics;

public class Kata
{
  public static BigInteger[] PowersOfTwo(int n)
  {
    return Enumerable.Range(0, n + 1).Select(i => BigInteger.Pow(2, i)).ToArray();
  }
}

// 3 Clever solution
using System.Linq;
using System.Numerics;

public class Kata
{
  public static BigInteger[] PowersOfTwo(int n) => Enumerable.Range(0, n + 1).Select(i => new BigInteger(1) << i).ToArray();
}
